packages/kbn-securitysolution-utils/src/path_validations/index.ts

```diff
@@ -0,0 +1,178 @@
+});
+
+export const FILEPATH_WARNING = i18n.translate('utils.filename.pathWarning', {
+  defaultMessage: `Path may be formed incorrectly; verify value`,
+});
+
+export enum ConditionEntryField {
+  HASH = 'process.hash.*',
+  PATH = 'process.executable.caseless',
+  SIGNER = 'process.Ext.code_signature',
+}
+
+export enum OperatingSystem {
+  LINUX = 'linux',
+  MAC = 'macos',
+  WINDOWS = 'windows',
+}
+
+type TrustedAppEntryTypes = 'match' | 'wildcard';
+/*
+ * regex to match executable names
+ * starts matching from the eol of the path
+ * file names with a single or multiple spaces (for spaced names)
+ * and hyphens and combinations of these that produce complex names
+ * such as:
+ * c:\home\lib\dmp.dmp
+ * c:\home\lib\my-binary-app-+/ some/  x/ dmp.dmp
+ * /home/lib/dmp.dmp
+ * /home/lib/my-binary-app+-\ some\  x\ dmp.dmp
+ */
+export const WIN_EXEC_PATH = /\\(\w+|\w*[\w+|-]+\/ +)+\w+[\w+|-]+\.*\w+$/i;
+export const UNIX_EXEC_PATH = /(\/|\w*[\w+|-]+\\ +)+\w+[\w+|-]+\.*\w*$/i;
+
+export const validateFilePathInput = ({
+  os,
+  value = '',
+}: {
+  os: OperatingSystem;
+  value?: string;
+}): string | undefined => {
+  const textInput = value.trim();
+  const isValidFilePath = isPathValid({
+    os,
+    field: 'file.path.text',
+    type: 'wildcard',
+    value: textInput,
+  });
+  const hasSimpleFileName = hasSimpleExecutableName({
+    os,
+    type: 'wildcard',
+    value: textInput,
+  });
+
+  if (!textInput.length) {
+    return FILEPATH_WARNING;
+  }
+
+  if (isValidFilePath) {
+    if (!hasSimpleFileName) {
+      return FILENAME_WILDCARD_WARNING;
+    }
+  } else {
+    return FILEPATH_WARNING;
+  }
+};
+
+export const hasSimpleExecutableName = ({
+  os,
+  type,
+  value,
+}: {
+  os: OperatingSystem;
+  type: TrustedAppEntryTypes;
+  value: string;
+}): boolean => {
+  if (type === 'wildcard') {
+    return os === OperatingSystem.WINDOWS ? WIN_EXEC_PATH.test(value) : UNIX_EXEC_PATH.test(value);
+  }
+  return true;
+};
+
+export const isPathValid = ({
+  os,
+  field,
+  type,
+  value,
+}: {
+  os: OperatingSystem;
+  field: ConditionEntryField | 'file.path.text';
+  type: TrustedAppEntryTypes;
+  value: string;
+}): boolean => {
+  if (field === ConditionEntryField.PATH || field === 'file.path.text') {
+    if (type === 'wildcard') {
+      return os === OperatingSystem.WINDOWS
+        ? isWindowsWildcardPathValid(value)
+        : isLinuxMacWildcardPathValid(value);
+    }
+    return doesPathMatchRegex({ value, os });
+  }
+  return true;
+};
+
+const doesPathMatchRegex = ({ os, value }: { os: OperatingSystem; value: string }): boolean => {
+  if (os === OperatingSystem.WINDOWS) {
+    const filePathRegex =
+      /^[a-z]:(?:|\\\\[^<>:"'/\\|?*]+\\[^<>:"'/\\|?*]+|%\w+%|)[\\](?:[^<>:"'/\\|?*]+[\\/])*([^<>:"'/\\|?*])+$/i;
+    return filePathRegex.test(value);
+  }
+  return /^(\/|(\/[\w\-]+)+|\/[\w\-]+\.[\w]+|(\/[\w-]+)+\/[\w\-]+\.[\w]+)$/i.test(value);
```

Comment:
Checked and this looks linear, 👍 :
<img width="790" alt="Screen Shot 2022-02-28 at 12 37 04 PM" src="https://user-images.githubusercontent.com/1151048/156047339-94ac9d3b-7f21-4934-9b1e-bdcfe3c73815.png">